src/main/java/Duke.java

```diff
@@ -1,10 +1,239 @@
+        fileStorage.deleteEntry(task);
+        return "Noted. I've removed task:\n  "
+                .concat(task.toString())
+                .concat("\nNow you have ".concat(taskList.size() + "").concat(" tasks in the list."));
+    }
+
+    // Level-3: Mark items as done
+    public String done(int index) {
+        taskList.get(index).setDone();
+        Task task = taskList.get(index);
+        return "Nice! I've marked task as done: \n  ".concat(task.toString());
+    }
+
+    // Level-2: Lists all items in Task List
+    public String list() {
+        String out = "Here are the tasks in your list:\n";
+        System.out.println(taskList);
+        for (int i = 0; i < taskList.size(); i++) {
+            if (i != 0) out = out.concat("\n");
+            out = out.concat((i+1) + ".").concat(taskList.get(i).toString());
+        }
+        return out;
+    }
+
+    // wraps all messages between line separators
+    public String messageWrapper(String text) {
+        return SEP_LINE.concat("\n").concat(text).concat("\n").concat(SEP_LINE);
+    }
+
+    // decodes input from user and passes that argument to a response builder
+    public int decoder(String userInput) {
+        int res;
+        switch (userInput) {
+        case "bye":
+            res = 0;
+            break;
+        case "list":
+            res = 1;
+            break;
+        case "done":
+            res = 2;
+            break;
+        case "todo":
+            res = 3;
+            break;
+        case "deadline":
+            res = 4;
+            break;
+        case "event":
+            res = 5;
+            break;
+        case "delete":
+            res = 6;
+            break;
+        default:
+            res = 7;
+            break;
+        }
+        return res;
+    }
+
+    // Level-5 -> A-Exceptions: Added try and except to handle errors
+    private void run() {
+        Scanner scannerObj = new Scanner(System.in);
+        greet();
+        while (isActive()) {
+            String nextIn;
+            try {
+                nextIn = scannerObj.next();
+            } catch (NoSuchElementException e) {
+                continue;
+            }
+            int selector = decoder(nextIn);
+            String output;
+            switch (selector) {
+            case 0:
+                output = exit();
+                break;
+            case 1:
+                output = list();
+                break;
+            case 2:
+                output = done(scannerObj.nextInt() - 1);
+                break;
+            case 3:
+                try {
+                    output = todo(scannerObj.nextLine());
+                    break;
+                } catch (NoSuchElementException e) {
+                    output = "☹ OOPS!!! The description of a todo cannot be empty.";
+                    break;
+                }
+            case 4:
+            case 5:
+                String desc = "";
+                String var;
```

Comment:
Perhaps a more specific variable name here as well, so that it is easier for readers to understand its purpose.